src/Altinn.Platform/Altinn.Platform.Events/Functions/Services/PushEventsService.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Altinn.Common.AccessTokenClient.Services;
+using Altinn.Platform.Events.Functions.Extensions;
+using Altinn.Platform.Events.Functions.Models;
+using Altinn.Platform.Events.Functions.Services.Interfaces;
+using Microsoft.Extensions.Logging;
+using Microsoft.IdentityModel.Tokens;
+
+namespace Altinn.Platform.Events.Functions.Services
+{
+    /// <summary>
+    /// Handles pushevents service
+    /// </summary>
+    public class PushEventsService : IPushEventsService
+    {
+        private readonly HttpClient _client;
+        private readonly IAccessTokenGenerator _accessTokenGenerator;
+        private readonly IKeyVaultService _keyVaultService;
+        private readonly ILogger<IPushEventsService> _logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PushEventsService"/> class.
+        /// </summary>
+        public PushEventsService(
+            HttpClient httpClient,
+            IAccessTokenGenerator accessTokenGenerator,
+            IKeyVaultService keyVaultService,
+            ILogger<IPushEventsService> logger)
+        {
+            httpClient.BaseAddress = new Uri(Environment.GetEnvironmentVariable("ApiPushEventsEndpoint"));
+            _client = httpClient;
+            _accessTokenGenerator = accessTokenGenerator;
+            _keyVaultService = keyVaultService;
+            _logger = logger;
+        }
+
+        /// <inheritdoc/>
+        public async Task SendToPushController(CloudEvent item)
+        {
+            StringContent httpContent = new StringContent(JsonSerializer.Serialize(item), Encoding.UTF8, "application/json");
+            try
+            {
+                string endpointUrl = "push";
+
+                SecurityTokenDescriptor tokenDescriptor = new SecurityTokenDescriptor();
+
+                string certBase64 = await _keyVaultService.GetSecretAsync(Environment.GetEnvironmentVariable("KeyVaultURI"), Environment.GetEnvironmentVariable("PlatformCertSecretId"));
```

Comment:
I would like to see the retrieval of the environment variables in the constructor.. They might be reused by other functions in the future I assume.

Reply:
Changed